Classes/common/CDTReplicator/CDTReplicatorFactory.m

```diff
@@ -45,9 +45,16 @@ - (id)initWithDatastoreManager:(CDTDatastoreManager *)dsManager
 {
     self = [super init];
     if (self) {
-        self.manager = dsManager;
-        TD_DatabaseManager *dbManager = dsManager.manager;
-        self.replicatorManager = [[TDReplicatorManager alloc] initWithDatabaseManager:dbManager];
+        
+        if(dsManager){
+            self.manager = dsManager;
```

Comment:
It is better not to call `self` inside `init`, instead use the ivars. There is a good explanation [here](http://qualitycoding.org/objective-c-init/).